typescript-test-samples/launchdarkly-lambda-dynamodb/readme.md

```diff
@@ -0,0 +1,90 @@
+
+---
+
+## Contents
+
+We've provided three example Lambdas to demonstrates the basics of deploying LaunchDarkly within AWS serverless.
+
+1. **getFlagsStarter** – This example uses the LaunchDarkly Node.js server SDK to demonstrate how to initialize the SDK client, get a flag value and, as a base use case, establish separate code paths that can be tested within your Lambda. The example uses an anonymous user context which could be used for randomized experimentation or percentage-based progressive rollouts. For more granular control of targeting for testing, you can pass an environment context or send Cognito user data.
+2. **getFlagsWithDynamo** – This takes the prior example but adds DynamoDB as a persistent feature store for LaunchDarkly feature flags enabling the SDK client to evaluate flag results based upon the data stored in DynamoDB. It is configured to enable daemon mode, which means that it will not call out to LaunchDarkly for these initial values, which can decrease initialization cost and reduce external requests. To disable daemon mode, remove the `useLdd` flag from the configuration.
+3. **syncFlagsToDynamo** – This is a simple Lambda designed to force cache flag values into DynamoDB enabling the use of daemon mode. While this solution works, it requires the function to be called (manually or programmatically) to start the initialization. The ideal solution for this problem is to use LauchDarkly's [Relay Proxy](https://docs.launchdarkly.com/home/relay-proxy/).
+
+For a complete guide to using LaunchDarkly in AWS serverless, including how to set up the Relay Proxy within your AWS environment and how to ensure that all events are captured by the SDKs within an ephemeral Lambda, visit [this guide](https://launchdarkly.com/blog/using-launchdarkly-in-aws-serverless/).
+
+The project also includes the code necessary to build and deploy a Lambda Layer to easily add the necessary LaunchDarkly dependencies to any Lambda that requires them. More on how to create and use the Layer below.
+
+## About This Example
+
+It can be difficult to fully test features in Lambdas outside of a production cloud environment. Feature flags can be a powerful pattern for enabling and disabling features for testing. LaunchDarkly's advanced targeting capabilities allow you to create feature flags that canbe used to safely test features in production without impacting the existing experience. Using feature flags and targeting, you can run code paths that are enabled for one user or one environment (or any combination of criteria) but not for anyone else. For example, utilizing authentication, environment and identity data, you can enable features for:
+
+* Only particular testing or staging environments
+* Only particular users or segments of users
+* Beta testing with managed beta access customers
+* Progressive rollouts to stress testing features
+* Experimentation on features or performance
+
+...and much more. 
+
+## The LaunchDarkly SDKs
```

Comment:
Good intro to the SDKs!

Can we be explicit about a LaunchDarkly account being needed (or not needed) to run the example? It may also be a good idea to add a link to LaunchDarkly "Getting Started" content if customers want to take the next step.